shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -340,7 +340,11 @@ - (instancetype)initWithEngine:(nonnull FlutterEngine*)engine
     }
     _engine = engine;
     CommonInit(self);
-    [engine setViewController:self];
```

Comment:
Bypassing the initialization logic by using `.viewController =` instead of `setViewController` seems very hacky. I'm surprised that `viewController` is not readonly (I suggest we make it so).

It feels to me that this PR complexes the relationship between `FVC` and `FlutterEngine`, which is indicated by having _three_ `initializeKeyboard` calls throughout the file. It's hard to track how many ways are there we expect to use `FVC` and whether a line is executed by each, as well as how to add a new line to the initialization.

Reply:
Commenting only about 
> Bypassing the initialization logic by using `.viewController =` instead of `setViewController` seems very hacky.

That's just Objective-C, they are completely identical under the covers.  The dot notation `.viewController =` is a more idiomatic way to set a property.

Reply:
> It feels to me that this PR complexes the relationship

It does. When the existing engine is passed, it can be running or not. So that increases the complexity of the initializer and the patch adds basically one `if` which loads `FlutterView` and initializes the keyboard if it's running. So to me the added complexity doesn't look excessive.

> It's hard to track how many ways are there we expect to use FVC and whether a line is executed by each, as well as how to add a new line to the initialization.

There is `CommonInit` and all initializers use it. `initWithEngine` handles the case when the engine is running and I could extract that into the method, but I couldn't come up with a good name and it doesn't clarify anything in that context.

Reply:
> _three_ initializeKeyboard calls

To me, the story with the keyboard is rather complicated. There are quite a few cases to handle, and I'm not sure controller is the best place for that. There is a `TODO` comment in `initializeKeyboard` which I guess, could solve some of the problems. But fixing that `TODO` goes way beyond my capabilities/knowledge of the engine's internals. 
1. Engine is not running, then `FVC` is created (displaying `FVC` will start the engine)
2. Engine is running, then `FVC` is created
3. Engine is not running, then `FVC` is created, and the engine is started by the user, and later `FVC` is shown (not handled by this patch, but `viewWillAppear` perhaps should check if the keyboard was correctly configured before)

One solution could be that the engine would fire an event when it's started, and `FVC` would subscribe to that. This can simplify keyboard initialization for cases 1. and 3.